ui/app/components/page/userpass-reset-password.js

```diff
@@ -0,0 +1,39 @@
+import { action } from '@ember/object';
+import { service } from '@ember/service';
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { task } from 'ember-concurrency';
+import errorMessage from 'vault/utils/error-message';
+
+export default class PageUserpassResetPasswordComponent extends Component {
+  @service store;
+
+  @tracked newPassword = '';
+  @tracked successful = false;
+  @tracked error = '';
+
+  @action reset() {
+    this.successful = false;
+    this.error = '';
+    this.newPassword = '';
+  }
+
+  @task
+  *updatePassword(evt) {
+    evt.preventDefault();
+    this.error = '';
+    const adapter = this.store.adapterFor('auth-method');
+    const { backend, username } = this.args;
+    if (!backend || !username) return;
```

Comment:
It seems like the form shouldn't render if there is no backend or username, is this just an added safety check?

Reply:
This is a safety check. Whether the form should render is handled on the parent